vault/resource_aws_auth_backend_roletag_blacklist.go

```diff
@@ -0,0 +1,153 @@
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"safety_buffer": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The amount of extra time that must have passed beyond the roletag expiration, before it's removed from backend storage.",
+			},
+			"disable_periodic_tidy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, disables the periodic tidying of the roletag blacklist entries.",
+			},
+		},
+	}
+}
+
+func awsAuthBackendRoleTagBlacklistWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("safety_buffer"); ok {
+		data["safety_buffer"] = v
+	}
+	if v, ok := d.GetOk("disable_periodic_tidy"); ok {
+		data["disable_periodic_tidy"] = v
+	}
+
+	path := awsAuthBackendRoleTagBlacklistPath(backend)
+
+	log.Printf("[DEBUG] Configuring AWS auth backend roletag blacklist %q", path)
+	_, err := client.Logical().Write(path, data)
+
+	d.SetId(path)
+
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("Error configuring AWS auth backend roletag blacklist %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Configured AWS backend roletag blacklist %q", path)
+
+	return awsAuthBackendRoleTagBlacklistRead(d, meta)
+}
+
+func awsAuthBackendRoleTagBlacklistRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Id()
+
+	backend, err := awsAuthBackendRoleTagBlacklistBackendFromPath(path)
+	if err != nil {
+		return fmt.Errorf("Invalid path %q for AWS auth backend roletag blacklist: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Reading roletag blacklist %q from AWS auth backend", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("Error reading AWS auth backend roletag blacklist %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Read roletag blacklist %q from AWS auth backend", path)
+	if resp == nil {
+		log.Printf("[WARN] AWS auth backend roletag blacklist %q not found, removing it from state", path)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("safety_buffer", resp.Data["safety_buffer"])
+	d.Set("disable_periodic_tidy", resp.Data["disable_periodic_tidy"])
+	d.Set("backend", backend)
+
+	return nil
+}
+
+func awsAuthBackendRoleTagBlacklistDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+
+	log.Printf("[DEBUG] Removing roletag blacklist %q from AWS auth backend", path)
+	_, err := client.Logical().Delete(path)
+	if err != nil {
+		return fmt.Errorf("Error deleting AWS auth backend roletag blacklist %q: %s", path, err)
```

Comment:
Will `Delete` return an error that tells us this resource already no longer exists? If so we can catch that error and simply log that and remove from state without this error. This probably isn't 100% required though because you've implemented the `Exists` method though... I suppose a follow up refresh/plan would remove it from that 🤔

Reply:
I just tried it with curl, and it appears DELETE is nilpotent. No matter how many times I call it--even if I never configured the blacklist--as long as the backend is actually enabled, the request succeeds.

Unfortunately, catching the 404 returned if the backend doesn't exist would be... tricky. Vault's error returns are basically big `fmt.Errorf` statements printing everything into human-readable formats. And while we _could_ try to parse out the error, it's going to be brittle.

That said, Exists won't catch this, unfortunately, as it'll run into the same problem. If it tries to check if the configuration exists, that works, but the way the Vault client works, I can't seem to find any way to detect when the backend is missing without resorting to string parsing.